src/bots/jitMaker.ts

```diff
@@ -280,447 +168,413 @@ export class JitMakerBot implements Bot {
+		} catch (e) {
+			logger.error(`Failed to send chunked placeOrder tx: ${e}`);
+			return;
 		}
+	}
 
-		const takerUserAccount = (
-			await this.userMap.mustGet(action.node.userAccount.toString())
-		).getUserAccount();
-		const takerAuthority = takerUserAccount.authority;
+	async doSpotHedgeTrades(
+		spotMarketIndex: number,
+		driftClient: DriftClient,
+		jupiterClient: JupiterClient,
+		tradeSize: BN,
+		maxDollarSize: BN,
+		direction: PositionDirection
+	): Promise<
+		| {
+				ixs: TransactionInstruction[];
+				lookupTables: AddressLookupTableAccount[];
+		  }
+		| undefined
+	> {
+		let jupSwapMode: SwapMode;
+		let tsize: BN;
+		let inMarketIndex: number;
+		let outMarketIndex: number;
+		// let jupReduceOnly: SwapReduceOnly;
+		if (isVariant(direction, 'long')) {
+			// sell USDC, buy spotMarketIndex
+			inMarketIndex = 0;
+			outMarketIndex = spotMarketIndex;
+			jupSwapMode = 'ExactIn';
+			tsize = maxDollarSize;
+			// jupReduceOnly = SwapReduceOnly.In;
+		} else {
+			// sell spotMarketIndex, buy USDC
+			inMarketIndex = spotMarketIndex;
+			outMarketIndex = 0;
+			jupSwapMode = 'ExactIn';
+			tsize = tradeSize;
+			// jupReduceOnly = SwapReduceOnly.In;
+		}
 
-		const takerUserStats = await this.userStatsMap.mustGet(
-			takerAuthority.toString()
-		);
-		const takerUserStatsPublicKey = takerUserStats.userStatsAccountPublicKey;
-		const referrerInfo = takerUserStats.getReferrerInfo();
-
-		return await this.driftClient.placeAndMakePerpOrder(
-			{
-				orderType: OrderType.LIMIT,
-				marketIndex: action.marketIndex,
-				baseAssetAmount: action.baseAssetAmount,
-				direction: action.direction,
-				price: action.price,
-				postOnly: PostOnlyParams.TRY_POST_ONLY,
-				immediateOrCancel: true,
-			},
-			{
-				taker: action.node.userAccount,
-				order: action.node.order,
-				takerStats: takerUserStatsPublicKey,
-				takerUserAccount: takerUserAccount,
-			},
-			referrerInfo
+		logger.info(
+			`Jupiter swap: ${getVariant(
+				direction
+			)}: ${tradeSize.toString()}, inMarket: ${inMarketIndex}, outMarket: ${outMarketIndex}, 
+			jupSwapMode: ${jupSwapMode}`
 		);
-	}
 
-	private async tryMakeJitAuctionForMarket(
-		market: PerpMarketAccount,
-		dlob: DLOB
-	) {
-		await this.updateAgentState();
-		await this.drawAndExecuteAction(market, dlob);
-	}
+		const inMarket = driftClient.getSpotMarketAccount(inMarketIndex);
+		const outMarket = driftClient.getSpotMarketAccount(outMarketIndex);
+		const routes = await jupiterClient.getRoutes({
+			inputMint: inMarket.mint,
+			outputMint: outMarket.mint,
+			amount: tsize,
+			swapMode: jupSwapMode,
+		});
 
-	private async tryMake() {
-		const start = Date.now();
-		let ran = false;
-		try {
-			await tryAcquire(this.periodicTaskMutex).runExclusive(async () => {
-				const dlob = this.dlobSubscriber.getDLOB();
-				if (!dlob) {
-					logger.info(`${this.name}: DLOB not initialized yet`);
-					return;
-				}
+		if (routes.length === 0) {
+			return undefined;
+		}
 
```

Comment:
not sure if we wanna bother, but this might be a good place to check the actual price of the swap route is not too ridiculous compared to oracle price

Reply:
will leave a TODO there